src/core/or/circuitpadding.c

```diff
@@ -2602,6 +2602,36 @@ circpad_handle_padding_negotiated(circuit_t *circ, cell_t *cell,
   return 0;
 }
 
+/** Free memory allocated by this machine spec. */
+STATIC void
+machine_spec_free_(circpad_machine_spec_t *m)
+{
+  if (!m) return;
+
+  tor_free(m->states);
+  tor_free(m);
+}
+
+/** Free all memory allocated by the circuitpadding subsystem. */
```

Comment:
We should comment somewhere that circpad machine_specs must all be malloced now, if this function is used.

I think our unittests actually add some non-malloc'd machines from global static allocation to the smartlist..

Reply:
Agreed. Added comments in 7dc7c02f13.